src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -998,5 +998,89 @@ static void RefParam(ref dynamic p)
                 expectedOutput: "frog",
                 additionalRefs: new[] { SystemCoreRef, CSharpRef }).VerifyDiagnostics();
         }
+
+        [Fact]
+        public void RefQueryClause()
```

Comment:
>RefQueryClause [](start = 20, length = 14)

Please test all query clauses that include expressions "converted" to lambdas: where, order by, join, etc.